tests/cpp/nodes/mathematical/test_binaryop.cpp

```diff
@@ -24,7 +24,11 @@
 
 namespace dwave::optimization {
 
-TEMPLATE_TEST_CASE("BinaryOpNode", "", std::equal_to<double>, std::less_equal<double>,
+#define CHECK_CLOSE_ENOUGH(a,b) CHECK(std::abs(a - b) < 10e-16)
+
+TEMPLATE_TEST_CASE("BinaryOpNode", "", 
+                    // std::divides<double>, 
```

Comment:
I'm having a bunch of troubles turning this on and I'm having troubles seeing output.  Question for @arcondello: Any quick thoughts on why this might be happening?  

It's failing in some asserts during niitialization:
```
    // check that all values are within min/max
    if (array_ptr->size(state)) {
        assert(std::ranges::min(array_ptr->view(state)) >= array_ptr->min());
        assert(std::ranges::max(array_ptr->view(state)) <= array_ptr->max());
        assert(!array_ptr->integral() || std::ranges::all_of(array_ptr->view(state), is_integer));
    }
```

Reply:
That is checking for consistency with the outputs of `max()`,`min()` and `integral()`. 
Which one is failing?

Reply:
```[13/13] Linking target tests/cpp/tests-all
1/1 tests-all RUNNING       
>>> MSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 LD_LIBRARY_PATH=/home/mcoury/repos/github/mdcoury/dwave-optimization/build/::/opt/clp/dist/lib/:/opt/clp/dist/lib/ MESON_TEST_ITERATION=1 MALLOC_PERTURB_=55 ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 /home/mcoury/repos/github/mdcoury/dwave-optimization/build/tests/cpp/tests-all
――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――
tests-all: ../dwave/optimization/src/nodes/testing.cpp:75: virtual void dwave::optimization::ArrayValidationNode::initialize_state(dwave::optimization::State&) const: Assertion `std::ranges::min(array_ptr->view(state)) >= array_ptr->min()' failed.
Randomness seeded to: 2900905893

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tests-all is a Catch2 v3.7.1 host application.
Run with -? for options

-------------------------------------------------------------------------------
BinaryOpNode - std::divides<double>
      Given: Two constant arrays operated on
       When: We make a state
-------------------------------------------------------------------------------
../tests/cpp/nodes/mathematical/test_binaryop.cpp:90
...............................................................................

../tests/cpp/nodes/mathematical/test_binaryop.cpp:90: FAILED:
due to a fatal error condition:
  SIGABRT - Abort (abnormal termination) signal

===============================================================================
test cases:    3 |    2 passed | 1 failed
assertions: 1022 | 1021 passed | 1 failed

――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
1/1 tests-all FAIL            0.09s   killed by signal 6 SIGABRT


Ok:                 0   
Expected Fail:      0   
Fail:               1   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   

Full log written to /home/mcoury/repos/github/mdcoury/dwave-optimization/build/meson-logs/testlog.txt
```

Reply:
```==================================== 1/1 =====================================
test:         tests-all
start time:   17:51:45
duration:     0.09s
result:       killed by signal 6 SIGABRT
command:      MSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 LD_LIBRARY_PATH=/home/mco
ury/repos/github/mdcoury/dwave-optimization/build/::/opt/clp/dist/lib/:/opt/clp/dist/lib/ MESON_TEST_ITERATION=1 MALLOC_
PERTURB_=55 ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1
:print_summary=1:print_stacktrace=1 /home/mcoury/repos/github/mdcoury/dwave-optimization/build/tests/cpp/tests-all
----------------------------------- stdout -----------------------------------
Randomness seeded to: 2900905893

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tests-all is a Catch2 v3.7.1 host application.
Run with -? for options

-------------------------------------------------------------------------------
BinaryOpNode - std::divides<double>
      Given: Two constant arrays operated on
       When: We make a state
-------------------------------------------------------------------------------
../tests/cpp/nodes/mathematical/test_binaryop.cpp:90
...............................................................................

../tests/cpp/nodes/mathematical/test_binaryop.cpp:90: FAILED:
due to a fatal error condition:
  SIGABRT - Abort (abnormal termination) signal

===============================================================================
test cases:    3 |    2 passed | 1 failed
assertions: 1022 | 1021 passed | 1 failed

----------------------------------- stderr -----------------------------------
tests-all: ../dwave/optimization/src/nodes/testing.cpp:75: virtual void dwave::optimization::ArrayValidationNode::initia
lize_state(dwave::optimization::State&) const: Assertion `std::ranges::min(array_ptr->view(state)) >= array_ptr->min()' 
failed.
==============================================================================


Summary of Failures:

1/1 tests-all FAIL            0.09s   killed by signal 6 SIGABRT

Ok:                 0   
Expected Fail:      0   
Fail:               1   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   
```

Reply:
Can we add a comment here? Even just a `todo: update to handle zero divisor case` or whatever

Reply:
I added a comment in the test_binaryop.cpp.  Shall I add another to the ArrayValidationNode?

Reply:
Nah, just in the tests IMO